ROX-25216: S3 Compatible Backup Integration (#78737)
openshift/openshift-docs: modules/s3-api-compatible-services-configuring-acs.adoc

// Module included in the following assemblies:
//
// * integration/integrate-with-s3-api-compatible-services.adoc
:_mod-docs-content-type: PROCEDURE
[id="s3-api-compatible-services-configuring-acs_{context}"]
= Configuring S3 API compatible integrations in {product-title}

To configure S3 API compatible backups, create a new integration in {product-title}.

.Prerequisites

> * You have configured an existing S3 bucket. +

Reviewer: ```suggestion
* You have configured an existing S3 bucket. 
```

// Module included in the following assemblies:
//
// * integration/integrate-with-s3-api-compatible-services.adoc
:_mod-docs-content-type: PROCEDURE
[id="s3-api-compatible-services-configuring-acs_{context}"]
= Configuring S3 API compatible integrations in {product-title}

To configure S3 API compatible backups, create a new integration in {product-title}.

.Prerequisites

> * An existing S3 Bucket. +

Reviewer: ```suggestion
* You have configured an existing S3 bucket. +
```

To create a new bucket with required permissions, see your S3 provider documentation.

> * `Read`, `write`, and `delete` permissions for the S3 bucket, the *Access key ID*, and the *Secret access key*.

Reviewer: ```suggestion
* You have `read`, `write`, and `delete` permissions for the S3 bucket, the *Access key ID*, and the *Secret access key*.
```

Reviewer: Normally the prereqs are complete sentences.

.Procedure
. In the {product-title-short} portal, go to *Platform Configuration* -> *Integrations*.
. Scroll down to the *External backups* section and select *S3 API Compatible*.
. Click *New Integration*.
. Enter a name for *Integration Name*.
. Enter the number of backups to retain in the *Backups To Retain* box.

> . For *Schedule*, select the backup frequency as daily or weekly as well as the time to run the backup process.

Reviewer: ```suggestion
. For *Schedule*, select the backup frequency as daily or weekly, and select the time to run the backup process.
```
Not sure if you select the time or enter it since I don't have access to the GUI, so use whichever is appropriate.

Author: You select the time.

. Enter the *Bucket* name where you want to store the backup.
. Optionally, enter an *Object Prefix* if you want to save the backups in a specific folder structure.
. Enter the *Endpoint* under which the S3 compatible service is reached. +

> Note that it must be reachable via https.

Reviewer: If you want this to be a note, we have a note object. Otherwise I would remove the "Note" wording and add this as a sentence right after "Enter the endpoint...."

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#admonitions

example of a note:

![Screenshot 2024-07-15 at 2 44 59 PM](https://github.com/user-attachments/assets/f4063110-d708-456f-89de-ae71135f2c53)

Author: I've removed `Note` here: [41ddcf2](https://github.com/openshift/openshift-docs/pull/78737/commits/41ddcf271379279d3776ccaebedeeb2efc1eb3df)

. Enter the *Region* for the bucket. +

> Consult your provider's documentation to enter the correct region.

Reviewer: I would combine this with the previous sentence. We don't usually add a line below the numbered step like this.

Author: 👍 I've changed it here: [49995fb](https://github.com/openshift/openshift-docs/pull/78737/commits/49995fbb8fdb5b991bb9a77095d78db0f37d694a)

> . Select the *URL style*. +

Reviewer: ```suggestion
. Select the *URL style*:
```
You don't need the `+` - if you use asterisks for the next 2 choices, the formatting should automatically be correct and show them under the numbered step.

> *Virtual hosted* style buckets are addressed as __\https://<bucket>.<endpoint>__. +

Reviewer: ```suggestion
* *Virtual hosted* style buckets are addressed as __\https://<bucket>.<endpoint>__. +
```

> *Path* style buckets are addressed as __\https://<endpoint>/<bucket>__.

Reviewer: ```suggestion
* *Path* style buckets are addressed as __\https://<endpoint>/<bucket>__.
```

> . Enter the *Access Key ID*, and the *Secret Access Key*.

Reviewer: ```suggestion
. Enter the *Access Key ID* and the *Secret Access Key*.
```

. Select *Test* to confirm that the integration with the S3 is working.
. Select *Create* to create the configuration.

> Once configured, {product-title} automatically backs up all data according to the specified schedule.

Reviewer: ```suggestion
After the integration is configured, {product-title-short} automatically backs up all data according to the specified schedule.
```
IBM Style: [do not use once](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#o) - "Use only to mean one time. Do not use as a conjunction to mean after or when."
